### PR TITLE
add support for windows and fix deprecated macros

### PR DESCRIPTION
**Pull request by talonateox**

### CODE REVIEW

File: windows-x64/SDL3-static.lib
Comment (drakbar) [inline windows-x64/SDL3-static.lib:1]: This is a binary target, which is specific to your machine. Binary files such as these should not be committed to source code repositories, it belongs in an build artifact repo. Furthermore, it is 10 MB of unnecessary bloat if you are a developer some where in the world with a limited internet connection targeting linux. Long story short, this file is not appropriate to be included as a part of this repo.

Comment (talonateox) [reply windows-x64/SDL3-static.lib:1]: the static lib is targeted to windows-x64, it should work on anyones machine running 64 bit windows, and i can change it to be the dynlib? its 1mb iirc

Comment (drakbar) [reply windows-x64/SDL3-static.lib:1]: #### **No build artifacts in the repo**. 
It is not only a matter of maintaining a minimal footprint, but also security. Leave the process of meeting dependency requirements to the user of the code. If you would like the directory `windows-x64` to remain in place you can drop a blank `.gitkeep` file inside it and push it up.

**Review by drakbar — APPROVED** [review]
Looks good.

**Pull request merged**